compose/compose-file.md

```diff
@@ -770,20 +932,39 @@ If you do not use a host path, you may specify a `volume_driver`.
 
     volume_driver: mydriver
 
-Note that for [version 2 files](compose-file.md#version-2), this driver
-will not apply to named volumes (you should use the `driver` option when
-[declaring the volume](compose-file.md#volume-configuration-reference) instead).
-For [version 1](compose-file.md#version-1), both named volumes and container volumes will
-use the specified driver.
+There are several things to note, depending on which
+[Compose file version](compose-file.md#versioning) you're using:
+
+-   `volume_driver` is not supported at all in
+    [version 3](compose-file.md#version-3). Instead of setting the volume driver
+    on the service, define a volume using the
+    [top-level `volumes` option](compose-file.md#volume-configuration-reference)
+    and specify the driver there.
+
+-   You can use `volume_driver` in [version 2 files](compose-file.md#version-2),
+    but it will only apply to anonymous volumes (those specified in the image,
+    or specified under `volumes` without an explicit named volume or host path).
+    To configure the driver for a named volume, use the `driver` key under the
+    entry in the
+    [top-level `volumes` option](compose-file.md#volume-configuration-reference).
+
+-   For [version 1 files](compose-file.md#version-1), both named volumes and
+    container volumes will use the specified driver.
 
-> Note: No path expansion will be done if you have also specified a
-> `volume_driver`.
+-   No path expansion will be done if you have also specified a `volume_driver`.
+    For example, if you specify a mapping of `./foo:/data`, the `./foo` part
+    will be passed straight to the volume driver without being expanded.
```

Comment:
Yep, check out `test_home_directory_with_driver_does_not_expand` in `config_test.py`